openscm/core/parameterset.py

```diff
@@ -215,17 +216,37 @@ def timeseries(  # pylint: disable=too-many-arguments
         ValueError
             Name not given or invalid region
         """
-        timeseries_type = ParameterType.from_timeseries_type(timeseries_type)
-        interpolation = InterpolationType.from_interpolation_type(interpolation)
-        extrapolation = ExtrapolationType.from_extrapolation_type(extrapolation)
         parameter = self._get_or_create_parameter(
             name, self._get_or_create_region(region)
         )
 
-        parameter.attempt_read(timeseries_type, unit, time_points)
-        return TimeseriesView(
-            parameter, unit, time_points, timeseries_type, interpolation, extrapolation
-        )  # TimeseriesView
+        if parameter.unit is not None:
+            # check conversion can be done
+            UnitConverter(parameter.unit, unit)
```

Comment:
Should not be necessary, is done in `TimeseriesView` (which should defenitely be returned in any case)

Reply:
As above, I'll need to modify TimeseriesConverter so it can take time points of none or modify `TimseriesView` so it just doesn't have a `TimeseriesConverter` if `time_points` is `None`.